qa/tasks/workunit.py

```diff
@@ -321,6 +321,7 @@ def _run_tests(ctx, refspec, role, tests, env, subdir=None, timeout=None):
                 run.Raw('&&'),
                 'git',
                 'clone',
+                '--depth=1',
```

Comment:
we could, but the time is totally dominated by the remote pack creation and transmission.  I can't observe any difference at all from --no-checkout.